libs/features/creature/src/creature-text/creature-text.integration.spec.ts

```diff
@@ -0,0 +1,280 @@
+import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
+import { RouterTestingModule } from '@angular/router/testing';
+import { MysqlQueryService, SqliteService } from '@keira/shared/db-layer';
+import { MultiRowEditorPageObject, TranslateTestingModule } from '@keira/shared/test-utils';
+import { CreatureText } from '@keira/shared/acore-world-model';
+import { ModalModule } from 'ngx-bootstrap/modal';
+import { ToastrModule } from 'ngx-toastr';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { of } from 'rxjs';
+import { CreatureHandlerService } from '../creature-handler.service';
+import { SaiCreatureHandlerService } from '../sai-creature-handler.service';
+import { CreatureTextComponent } from './creature-text.component';
+import Spy = jasmine.Spy;
+import { instance, mock } from 'ts-mockito';
+class CreatureTextPage extends MultiRowEditorPageObject<CreatureTextComponent> {}
+
+describe('CreatureText integration tests', () => {
+  let fixture: ComponentFixture<CreatureTextComponent>;
+  let queryService: MysqlQueryService;
+  let querySpy: Spy;
+  let handlerService: CreatureHandlerService;
+  let page: CreatureTextPage;
+
+  const id = 1234;
+
+  const originalRow0 = new CreatureText();
+  const originalRow1 = new CreatureText();
+  const originalRow2 = new CreatureText();
+  originalRow0.CreatureID = originalRow1.CreatureID = originalRow2.CreatureID = id;
+  originalRow0.ID = 0;
+  originalRow1.ID = 1;
+  originalRow2.ID = 2;
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        BrowserAnimationsModule,
+        ToastrModule.forRoot(),
+        ModalModule.forRoot(),
+        CreatureTextComponent,
+        RouterTestingModule,
+        TranslateTestingModule,
+      ],
+      providers: [CreatureHandlerService, SaiCreatureHandlerService, { provide: SqliteService, useValue: instance(mock(SqliteService)) }],
+    }).compileComponents();
+  }));
+
+  function setup(creatingNew: boolean) {
+    handlerService = TestBed.inject(CreatureHandlerService);
+    handlerService['_selected'] = `${id}`;
+    handlerService.isNew = creatingNew;
+
+    queryService = TestBed.inject(MysqlQueryService);
+    querySpy = spyOn(queryService, 'query').and.returnValue(of([]));
+    spyOn(queryService, 'queryValue').and.returnValue(of());
+
+    spyOn(queryService, 'selectAll').and.returnValue(of(creatingNew ? [] : [originalRow0, originalRow1, originalRow2]));
+
+    fixture = TestBed.createComponent(CreatureTextComponent);
+    page = new CreatureTextPage(fixture);
+    fixture.autoDetectChanges(true);
+    fixture.detectChanges();
+  }
+
+  describe('Creating new', () => {
+    beforeEach(() => setup(true));
```

Comment:
this `beforeEach` will become obsolete with the new `setup` structure suggested in my comment above